src/vtbackend/Terminal.cpp

```diff
@@ -1487,7 +1486,6 @@ void Terminal::setLeftRightMargin(optional<ColumnOffset> _left, optional<ColumnO
         {
             state_.margin.horizontal.from = left;
             state_.margin.horizontal.to = right;
```

Comment:
With this change and the one above you should apply the following to not break DECSTBM / DECSLRM:

```diff
diff --git a/src/vtbackend/Screen.cpp b/src/vtbackend/Screen.cpp
index 063ba91a..0bfa038e 100644
--- a/src/vtbackend/Screen.cpp
+++ b/src/vtbackend/Screen.cpp
@@ -3461,6 +3461,7 @@ ApplyResult Screen<Cell>::apply(FunctionDefinition const& function, Sequence con
             auto l = decr(seq.param_opt<ColumnOffset>(0));
             auto r = decr(seq.param_opt<ColumnOffset>(1));
             _terminal.setLeftRightMargin(l, r);
+            moveCursorTo({}, {});
         }
         break;
         case DECSM: {
@@ -3474,6 +3475,7 @@ ApplyResult Screen<Cell>::apply(FunctionDefinition const& function, Sequence con
         case DECSTBM:
             _terminal.setTopBottomMargin(decr(seq.param_opt<LineOffset>(0)),
                                          decr(seq.param_opt<LineOffset>(1)));
+            moveCursorTo({}, {});
             break;
         case DECSTR:
             // For VTType VT100 and VT52 ignore this sequence
```

We don't have an explicit test case for both VT sequences, because I **thought** we're using the implementation function a lot already. but now it turns out that we probably should have that (not part of this PR, necessarily :) )